gloo/allgather_ring.h

```diff
@@ -55,6 +55,10 @@ class AllgatherRing : public Algorithm {
   virtual ~AllgatherRing() {}
 
   void run() {
+    // Short circuit if there is only a single process or the output is empty.
+    if (this->contextSize_ == 1 || count_ == 0) {
```

Comment:
Why `contextSize_` needs `this->`, but not `count_`?

Reply:
That also struck me as unusual but I followed existing code that does it the same way, e.g. at line 71 in the original version